File: 백준/Silver/1874. 스택 수열/스택 수열.swift
var array = [Int]()
var result = ""
var now = 0
let N = Int(readLine()!)!
for _ in 0..<N {
    let input = Int(readLine()!)!
    if result == "NO" {
        continue
    }
    if now <= input {
        for i in now+1...input {
            array.append(i)
            result += "+\n"
        }
        now = array.removeLast()
        result += "-\n"
    } else {
        if array.last! == input {
            array.removeLast()
            result += "-\n"
        } else {
            result = "NO"
        }
    }
}
print(result)

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

input = sys.stdin.readline

n = int(input())
stack = []
result = []
next_num = 1
possible = True

for _ in range(n):
    x = int(input())
    while next_num <= x:
        stack.append(next_num)
        result.append('+')
        next_num += 1

    if not stack or stack[-1] != x:
        possible = False
        break

    stack.pop()
    result.append('-')

print('\n'.join(result) if possible else 'NO')